i2c/pcf8523_i2c/pcf8523_i2c.c

```diff
@@ -0,0 +1,162 @@
+
+    uint8_t alarm_val[4] = {0x01, 0x80, 0x80, 0x80};
+    // Write alarm values to registers
+    for (int i = 10; i < 14; ++i){
+        buf[0] = (uint8_t) i;
+        buf[1] = alarm_val[i - 10];
+        i2c_write_blocking(i2c_default, addr, buf, 2, false);
+    }
+}
+#endif
+
+void pcf8520_check_alarm(){
+    // Check bit 3 of control register 2 for alarm flags
+    uint8_t status[1];
+    uint8_t val = 0x01;
+    i2c_write_blocking(i2c_default, addr, &val, 1, true); // true to keep master control of bus
+    i2c_read_blocking(i2c_default, addr, status, 1, false);
+
+    if ((status[0] & 0x08)==0x08){
+        printf("ALARM RINGING");
+    } else {
+        printf("Alarm not triggered yet");
+    }
+}
+
+void pcf8520_convert_time(int conv_time[7],uint8_t raw_time[7]){
+    // Convert raw data into time
+    conv_time[0]= (10*(int)((raw_time[0] & 0x70) >> 4)) + ((int)(raw_time[0] & 0x0F));
+    conv_time[1]= (10*(int)((raw_time[1] & 0x70) >> 4)) + ((int)(raw_time[1] & 0x0F));
+    conv_time[2]= (10*(int)((raw_time[2] & 0x30) >> 4)) + ((int)(raw_time[2] & 0x0F));
+    conv_time[3]= (10*(int)((raw_time[3] & 0x30) >> 4)) + ((int)(raw_time[3] & 0x0F));
+    conv_time[4]= (int)(raw_time[4] & 0x07);
+    conv_time[5]= (10*(int)((raw_time[5] & 0x10) >> 4)) + ((int)(raw_time[5] & 0x0F));
+    conv_time[6]= (10*(int)((raw_time[6] & 0xF0) >> 4)) + ((int)(raw_time[6] & 0x0F));
+}
+
+int main() {
+    stdio_init_all();
+#if !defined(i2c_default) || !defined(PICO_DEFAULT_I2C_SDA_PIN) || !defined(PICO_DEFAULT_I2C_SCL_PIN)
+    #warning i2c/pcf8520_i2c example requires a board with I2C pins
+    puts("Default I2C pins were not defined");
+#else
+    printf("Hello, PCF8520! Reading raw data from registers...\n");
+
+    // This example will use I2C0 on the default SDA and SCL pins (4, 5 on a Pico)
+    i2c_init(i2c_default, 400 * 1000);
+    gpio_set_function(PICO_DEFAULT_I2C_SDA_PIN, GPIO_FUNC_I2C);
+    gpio_set_function(PICO_DEFAULT_I2C_SCL_PIN, GPIO_FUNC_I2C);
+    gpio_pull_up(PICO_DEFAULT_I2C_SDA_PIN);
+    gpio_pull_up(PICO_DEFAULT_I2C_SCL_PIN);
+    // Make the I2C pins available to picotool
+    bi_decl(bi_2pins_with_func(PICO_DEFAULT_I2C_SDA_PIN, PICO_DEFAULT_I2C_SCL_PIN, GPIO_FUNC_I2C));
+
+    pcf8520_reset();
+
+    pcf820_write_current_time();
+    pcf8520_set_alarm();
+    pcf8520_check_alarm();
+
+    uint8_t raw_time[7];
+    int real_time[7];
+    char days_of_week[7][12] = {"Sunday","Monday","Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"};
+
+    while (1) {
+        
+        pcf8520_read_raw(raw_time);
+        pcf8520_convert_time(real_time, raw_time);
+
+        printf("Time: %02d : %02d : %02d\n",real_time[2],real_time[1],real_time[0]);
+        printf("Date: %s %02d / %02d / %02d\n",days_of_week[real_time[4]], real_time[3], real_time[5], real_time[6]);
+        pcf8520_check_alarm();
+        
+        
+        sleep_ms(500);
+
+        // Clear terminal 
+        printf("\e[1;1H\e[2J");
```

Comment:
I dunno if it's "safe" to assume that all serial terminals will understand these escape-codes? :computer: :shrug: